discojs/src/aggregator/mean.ts

```diff
@@ -108,7 +108,6 @@ export class MeanAggregator extends Aggregator<WeightsContainer> {
 
     this.contributions = this.contributions.setIn([0, nodeId], contribution);
```

Comment:
Not related to your changes but why are we storing the history of previous contributions rather than only the current one? Seems unnecessary and inefficient

Reply:
yeah, I totally agree. the issue is that this mean aggregator does more than it should, combining two features: doing a mean of weights & enabling to use weights from other rounds. we should split it. and I think that the second feature shouldn't even be in an aggregator but the responsability of the user of the aggregator (`Client` currently).
I tried to rethink a bit how such aggregation should happen, here are a few notes:
* `makePayload` and `communicationRound` are very weird, we need to get rid of it
* secure aggregation can also work across rounds, we just need to tell other nodes which weights we wants
* we need to have a clear(er) way to synchronize nodes
  * in federated, that's trivial: the server sends a `NextRound { round: number, weights: WeightContainer }` message everytime, with the aggregation of the previous round
  * in decentralized, we can use the server to orchestrate the rounds but without the weights contained in the message